ts/src/pro/xt.ts

```diff
@@ -0,0 +1,1062 @@
+        const marketId = this.safeString2 (order, 's', 'symbol');
+        if (marketId !== undefined) {
+            const tradeType = ('symbol' in order) ? 'contract' : 'spot';
+            const market = this.safeMarket (marketId, undefined, undefined, tradeType);
+            const symbol = market['symbol'];
+            const orderId = this.safeString (order, 'i', 'orderId');
+            const previousOrders = this.safeValue (orders.hashmap, symbol, {});
+            const previousOrder = this.safeValue (previousOrders, orderId);
+            if (previousOrder === undefined) {
+                const parsed = this.parseWsOrder (order, market);
+                orders.append (parsed);
+                client.resolve (orders, 'order:' + tradeType);
+            } else {
+                const trade = this.parseWsOrderTrade (order);
+                if (previousOrder['trades'] === undefined) {
+                    previousOrder['trades'] = [];
+                }
+                previousOrder['trades'].push (trade);
+                previousOrder['lastTradeTimestamp'] = trade['timestamp'];
+                let totalCost = '0';
+                let totalAmount = '0';
+                const trades = previousOrder['trades'];
+                for (let i = 0; i < trades.length; i++) {
+                    const trade = trades[i];
+                    totalCost = Precise.stringAdd (totalCost, this.numberToString (trade['cost']));
+                    totalAmount = Precise.stringAdd (totalAmount, this.numberToString (trade['amount']));
+                }
+                if (Precise.stringGt (totalAmount, '0')) {
+                    previousOrder['average'] = Precise.stringDiv (totalCost, totalAmount);
+                }
+                previousOrder['cost'] = totalCost;
+                if (previousOrder['filled'] !== undefined) {
+                    previousOrder['filled'] = Precise.stringAdd (previousOrder['filled'], this.numberToString (trade['amount']));
+                    if (previousOrder['amount'] !== undefined) {
+                        const stringAmount = this.numberToString (previousOrder['amount']);
+                        const stringFilled = this.numberToString (previousOrder['filled']);
+                        previousOrder['remaining'] = Precise.stringSub (stringAmount, stringFilled);
+                    }
+                }
+                if (previousOrder['fee'] === undefined) {
+                    previousOrder['fee'] = {
+                        'rate': undefined,
+                        'cost': '0',
+                        'currency': this.numberToString (trade['fee']['currency']),
+                    };
+                }
+                if ((previousOrder['fee']['cost'] !== undefined) && (trade['fee']['cost'] !== undefined)) {
+                    const stringOrderCost = this.numberToString (previousOrder['fee']['cost']);
+                    const stringTradeCost = this.numberToString (trade['fee']['cost']);
```

Comment:
@samgermain why do we need all of this custom logic and manual order manipulation?

Reply:
> @samgermain why do we need all of this custom logic and manual order manipulation?

This is the same logic that was in krakenfutures, I was going to remove it and add a unified method on exchange after this PR, I've discussed it on other PRs but I can't find the conversation right now, I originally copied it from the coinbasepro exchange, but it basically keeps track of all the trades associated with an order

Reply:
@samgermain Ok but we can't just copy and paste, please make sure you test everything thoroughly

Reply:
I removed it, it wasn't good for this exchange